src/main/java/de/tum/in/www1/artemis/web/rest/GradeStepResource.java

```diff
@@ -84,22 +84,26 @@ public GradeStepResource(GradingScaleRepository gradingScaleRepository, GradeSte
         log.debug("REST request to get all grade steps for exam: {}", examId);
         User user = userRepository.getUserWithGroupsAndAuthorities();
         Course course = courseRepository.findByIdElseThrow(courseId);
+        Exam exam = examRepository.findByIdElseThrow(examId);
         authCheckService.checkHasAtLeastRoleInCourseElseThrow(Role.STUDENT, course, user);
         GradingScale gradingScale = gradingScaleRepository.findByExamIdOrElseThrow(examId);
         boolean isInstructor = authCheckService.isAtLeastInstructorInCourse(course, user);
-        if (!isInstructor && !gradingScale.getExam().resultsPublished()) {
+        if (!isInstructor && !exam.resultsPublished()) {
             return forbidden();
         }
-        GradeStepsDTO gradeStepsDTO = prepareGradeStepsDTO(gradingScale);
+        GradeStepsDTO gradeStepsDTO = prepareGradeStepsDTO(gradingScale, exam.getMaxPoints());
         return ResponseEntity.ok(gradeStepsDTO);
     }
 
-    private GradeStepsDTO prepareGradeStepsDTO(GradingScale gradingScale) {
+    private GradeStepsDTO prepareGradeStepsDTO(GradingScale gradingScale, int maxPoints) {
```

Comment:
question: isn't it `Integer` as declared in `src/main/java/de/tum/in/www1/artemis/domain/Course.java` and in `src/main/java/de/tum/in/www1/artemis/web/rest/dto/GradeStepsDTO.java`

Reply:
because the getMaxPoints in the Exam.java returns an int (it's apparently no longer optional). However, it's optional in the Course, hence the Integer there. I kept the Integer in the GradeStepsDTO since I will reuse it, in my next PR, where the equivalent Course endpoint would return the GradeStepsDTO. Since it can be optional for the course I opted to have it as optional in the DTO in general and just always set it for the Exam.

Reply:
ah ok, it is obviously not related to the functionality provided in that PR but I did not 100% get your explanation... why would it depend on a value being optional? One could also change the return type of the `getMaxPoints()` in `Exam.java` to Integer.

Reply:
One could do that, true. However, since it's not an optional value anymore, it doesn't make sense for it to return an Integer. Therefore, I left it as int.